packages/template-retail-react-app/app/components/badge-group/index.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import PropTypes from 'prop-types'
+import {useStyleConfig, Box, Badge} from '@salesforce/retail-react-app/app/components/shared/ui'
+
+const BadgeGroup = (props) => {
+    const styles = useStyleConfig('BadgeGroup')
+
+    if (!props.badgeLabels) {
+        return
+    }
+    //TODO: check for duplicate labels
+    // If product is not provided/undefined then render the provided badge labels. This is useful if user already has the logic to get the badge labels outside the badge group
+    let filteredLabels = props.badgeLabels
```

Comment:
I find the logic here a bit confusing since badgeLabels will be overridden when product is available? Why don't we turn this component to be a "dumb" one, meaning it will render whatever `badgeLabels` are? Then we separate the logic to get the badge in it owns function. E.g

```
const productLabels = getBadgeInfo(product) => {
    const filterLabels = product?.filter() || PRODUCT_TILE_BADGE_LABELS
   return filterLabels
}


// product-tile.js
const badgeLabels = getBadgeInfo()

<BadgeGroup badgeLabels={badgeLabels} />
```

Reply:
The purpose was to modularize the logic. This Badge has the product info, so it should be capable of figuring out the details. If the customer wants to change the Badge, they can just override badge with out touching the product tile